BlueprintUI/Sources/Environment/Environment.swift

```diff
@@ -34,33 +34,103 @@ import UIKit
 ///             }
 ///         }
 ///     }
-public struct Environment {
+public struct Environment: Equatable {
+
+    typealias OnDidRead = (StorageKey) -> Void
+
+    enum LayoutPass: Equatable {
+        case measurement
+        case layout
+    }
```

Comment:
We can make this change later if we decide it's worth it, but if we wanted to promote these changes to `MarketEnvironment` it would be nice if this was an extendible "scope" instead of being tied to measure and layout.

Reply:
Had the same thought; it should be easy enough to do this when we extract the environment, but for now I think this is sufficient.